src/replication.c

```diff
@@ -1741,3 +1265,124 @@ destroy_relication_mempool(void) {
 exit:
 	return rc;
 }
+
+static void
+destroy_resp_list(rcommon_cmd_t *rcomm_cmd)
+{
+	int i;
+
+	for (i = 0; i < rcomm_cmd->copies_sent; i++) {
+		if (rcomm_cmd->resp_list[i].data_ptr) {
+			free(rcomm_cmd->resp_list[i].data_ptr);
+		}
+	}
+}
+
+void *
+cleanup_deadlist(void *arg)
+{
+	spec_t *spec = (spec_t *)arg;
+	rcommon_cmd_t *rcomm_cmd;
+	int i, count = 0, entry_count = 0;
+
+	while (1) {
+		entry_count = get_num_entries_from_mempool(&spec->rcommon_deadlist);
+		while (entry_count) {
+			count = 0;
+			rcomm_cmd = get_from_mempool(&spec->rcommon_deadlist);
+
+			for (i = 0; i < rcomm_cmd->copies_sent; i++) {
+				if (rcomm_cmd->resp_list[i].status)
+					count++;
+			}
+
+			if (count == rcomm_cmd->copies_sent) {
+				destroy_resp_list(rcomm_cmd);
+				free(rcomm_cmd->iov[0].iov_base);
+
+				for (i=1; i<rcomm_cmd->iovcnt + 1; i++)
```

Comment:
caller of 'replicate', i.e., 'istgt_lu_disk_lbwrite' seems to be freeing this iov in error cases..
those frees in 'istgt_lu_disk_lbwrite' need to be avoided for REPLICATION macro